### DISCUSSION TITLE
Connect-PnPOnline w/ PS 7.2 & PnP 2.1.1 in Azure Runbook

### DISCUSSION THREAD

**Discussion by dremillard** [pnp/powershell#3088] [Q&A]
Hi all, 

I can successfully connect using Connect-PnPOnline with a cert and thumbprint in Azure Runbooks with PowerShell 5.1 and 7.1 using PnP.Powershell 1.12.   This is the typical move:

$AdminURL = "https://mytenant-admin.sharepoint.com"
$TenantID = "b3asdf75-cca6-4bbf-9eb6-b01asdf8e5cf"
$ThumbPrint = "FA6738420527962ASDF3F3FE79B0CE68410F610FC"
$ClientID = "59sdfedf-6aa0-4d28-a2d9-1b3dsssd3115"

$Conn = Connect-PnPOnline -Url $AdminURL -ClientId $ClientID -Tenant $TenantID -Thumbprint $ThumbPrint -ReturnConnection

Get-PnPTeamsTeam -Connection $Conn

But now I need some of the new capabilities in PnP.PowerShell 2.1.1 and the above no longer works.  Now, when I do the above move in an Azure Runbook using PowerShell 7.2 and PnP.PowerShell 2.1.1, it gives an error saying that it can't find a certificate with that thumbprint.  And this runbook is in the same Automation Account using the same Shared Resources (certificate) as the PnP 1.12 that is working.  

The actual error is:  "Cannot find certificate with this thumbprint in the certificate store".

Any thoughts on what could be causing this? Did the methodology for using a certificate in Shared Resources change with 2.1.1?

Thanks in advance for your help!

Dave

**Comment by AndersRask** [comment]
First off, you should probably obfuscate details like thumbprint or tenantid in here :) (if they are real)

I tried doing this on a Hybrid Worker runbook, and it worked fine with 5.1 in 1.12.1 and with 7.1 in 2.1.1
Are you running HWG or using an Azure runbook?

**Comment by dremillard** [comment]
Hi Anders, yeah those aren’t the actual values for appid and thumbprint!  And I’m running in Azure, not hybrid.  Thanks, Dave

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
